File: RQ3/C-REval-Desc/task03_desc.c
int* intersperse(int* numbers, int length, int delimeter, int* result_length) {
    if (length == 0) {
        *result_length = 0;
        return NULL;
    }

    int* result = (int*)malloc((length * 2 - 1) * sizeof(int));
    int index = 0;

    for (int i = 0; i < length - 1; i++) {
        result[index++] = numbers[i];
        result[index++] = delimeter;
    }

    result[index++] = numbers[length - 1];
    *result_length = index;

    return result;
}

File: RQ3/Java-REval-Desc/task03_desc.java
public class Translator {
    public static int[] intersperse(int[] numbers, int delimeter) {
        int[] _result;
        if (numbers.length == 0) {
            _result = new int[0];
        } else {
            int[] result = new int[numbers.length * 2 - 1];
            int index = 0;
            for (int n = 0; n < numbers.length - 1; n++) {
                result[index++] = numbers[n];
                result[index++] = delimeter;
            }
            result[index] = numbers[numbers.length - 1];
            _result = result;
        }
        assert func(_result); // POST: func is equivalent to: Insert a number 'delimeter' between every two consecutive elements of input list `numbers' >>> intersperse(new int[0], 4) new int[0] >>> intersperse(new int[]{1, 2, 3}, 4) new int[]{1, 4, 2, 4, 3}
    }
}